R/nested.R

```diff
@@ -88,3 +88,49 @@ create_filler_nested <- function(pd_nested) {
   select_(pd_nested, ~spaces, ~newlines, ~short, ~everything())
 }
 
+#' Serialize a nested parse table
+#'
+#' Helper function that recursively extracts terminals from a nested tibble.
+#' @param pd_nested A nested parse table.
+#' @param pass_indent Level of indention of a token.
+#' @return A character vector with all tokens in `pd_nested` plus whitespace,
+#'   line break and indention information added between the tokens.
+#' @importFrom purrr pmap
+serialize_parse_data_nested_helper <- function(pd_nested, pass_indent) {
+  out <- pmap(list(pd_nested$terminal, pd_nested$text, pd_nested$child,
+             pd_nested$spaces, pd_nested$newlines, pd_nested$indent),
+           function(terminal, text, child, spaces, newlines, indent) {
+             if (terminal) {
+               c(rep_char(" ", pass_indent), text,
+                 newlines_and_spaces(newlines, spaces))
+             } else {
+               c(serialize_parse_data_nested_helper(child, indent + pass_indent),
+                 newlines_and_spaces(newlines, spaces))
+             }
+        }
+  )
+  out
+}
+
+#' Serialize a nested parse table
+#'
+#' Collapses a nested parse table into its character vector representation. To
+#'   achieve this, obsolete white spaces that were inserted in
+#'   [serialize_parse_data_nested_helper] spaces before the tokens have to be
+#'   removed.
+#' @param pd_nested A nested parse table with line break, spaces and indention
+#'   information.
+#' @return A character string.
+serialize_parse_data_nested <- function(pd_nested) {
+  raw <- serialize_parse_data_nested_helper(pd_nested, pass_indent = 0) %>%
+    unlist()
+  newline <- which(raw == "\n")
+  token <- setdiff(1:length(raw), union(which(raw == ""),
```

Comment:
Ok, I tried that and it works. The question is how exactly it should be implemented. I see two possible approaches. 

* Use *lag_newlines* only (and drop *newlines*) and always inserting newlines and indention before a token, spaces after it.

* Use *lag_newlines* to determine whether there is indention before a token. Continue using newlines and spaces for inserting non-indention spacing after each token.

I think for now we can stick with the former. 
However, the latter might be preferable because it is more flexible in the sense that spacing information other than our computed indention can technically be stored. At the moment, we just set spacing after a token to zero when a line break comes after the token (see R/modify_pd.R, line 20). Maybe the flexibility is not needed later and it just complicates things, so I suggest to go with the former.